gnovm/pkg/gnolang/preprocess.go

```diff
@@ -390,6 +408,27 @@ var preprocessing atomic.Int32
 //   - Assigns BlockValuePath to NameExprs.
 //   - TODO document what it does.
 func Preprocess(store Store, ctx BlockNode, n Node) Node {
+	// XXX If initStaticBlocks doesn't happen here,
+	// XXX it means Preprocess on blocks might fail.
+	// XXX it works for now because preprocess also does pushInitBlock,
+	// XXX but it's kinda weird.
+	// XXX maybe consider moving initStaticBlocks here and ensure idempotency of it.
```

Comment:
remove "XXX" from the above 5 lines; the comment can stay but requires more thinking.

Reply:
Kept the original comment—it actually clearly explains the potential issues and possible future optimizations.